agent/agent.go

```diff
@@ -501,11 +484,10 @@ func (a *Agent) setupTask(agentCtx, setupCtx context.Context, tcInput *taskConte
 	tc.logger.System().Info("System logger initialized.")
 
 	if err := setupCtx.Err(); err != nil {
-		setupFailed = true
 		grip.Error(err)
 		tc.logger.Execution().Infof("Stopping task execution: %s", err)
 		shouldExit, err := a.handleTaskResponse(setupCtx, tc, evergreen.TaskSystemFailed, err.Error())
-		return tc, setupFailed, shouldExit, err
```

Comment:
This can also call `a.handleSetupError`.

Reply:
The complexity it would add to not 
`grip.Infof("Task complete: '%s'.", tc.task.ID)`
and instead `tc.logger.Execution().Infof("Stopping task execution: %s", err)` made it feel not worth it.

Reply:
Seems like this can use `handleSetupError` now that it logs both.